### PR TITLE
kubernetes_cluster: support specifying node_resource_group

### LINKED ISSUE #3354: [azurerm_kubernetes_cluster] - Add Option for Node Resource Group Name
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
With the [2019-04-15](https://github.com/Azure/AKS/blob/2019-04-15/CHANGELOG.md#release-04-15-2019) release of AKS support for custom resource group names was added. 

> * New Features
>    * Customers may now provide custom Resource Group names.
>       * This means that users are no longer locked into the MC_* resource name group. On cluster creation you may pass in a custom RG and AKS will inherit that RG, permissions and attach AKS resources to the customer provided resource group.
>       * Currently, you must pass in a new RG (resource group) must be new, and can not be a pre-existing RG. We are working on support for pre-existing RGs.
>       * This change requires newly provisioned clusters, existing clusters can not be migrated to support this new capability. Cluster migration across subscriptions and RGs is not currently supported.

This feature request is to have the custom resource group name field added to the terraform provider.

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration
I'm not completely clear if go-sdk actually offers this support yet but I wanted to get the issue out there to see if other folks also want this. Right now the [2019-04-30](https://github.com/Azure/azure-sdk-for-go/blob/master/services/containerservice/mgmt/2019-04-30/containerservice/models.go#L1731) version of the go-sdk has this node resource group property  marked as *READ-ONLY*. I could be reading this wrong though. The [CreateOrUpdate](https://github.com/Azure/azure-sdk-for-go/blob/master/services/containerservice/mgmt/2019-04-30/containerservice/managedclusters.go#L115) part of the client does reference the `2019-02-01` version of the rest API. That version of the [rest API](https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/createorupdate#managedcluster ) has the node resource as a configuration option.

As for syntax to add to azurerm_kubernetes_cluster, since it's a single response item with no dependencies I wouldn't expect this to need it's own block. Maybe that's better form though.

### References
* https://github.com/Azure/AKS/blob/2019-04-15/CHANGELOG.md#release-04-15-2019
* https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/createorupdate#managedcluster 
* https://docs.microsoft.com/en-us/azure/aks/faq#can-i-provide-my-own-name-for-the-aks-infrastructure-resource-group 
* https://github.com/Azure/azure-sdk-for-go/blob/master/services/containerservice/mgmt/2019-04-30/containerservice/models.go#L1731

### PR DESCRIPTION
**Pull request by invidian**
Closes #3354

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

### CODE REVIEW

**Comment by invidian** [comment]
>Thank you for the PR. Could we update the docs to reflect this change and add it to a test? Thanks slightly_smiling_face

I won't have time next week to take care after that, but I will do it week after. Docs should be easy, but could you suggest some implementation for tests?

**Comment by katbyte** [comment]
@invidian,

All you need to do is Copy and rename a test and then set the property, or if there is a `complete` test set it there 🙂

**Comment by invidian** [comment]
>should change of node_resource_group recreate the cluster?

[AKS FAQ](https://docs.microsoft.com/en-us/azure/aks/faq) answers that, so I added `ForceNew: true` and mentioned that in documentation.

File: azurerm/resource_arm_kubernetes_cluster_test.go [hunk 1]
```diff
@@ -1744,3 +1767,32 @@ resource "azurerm_kubernetes_cluster" "test" {
...
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  dns_prefix          = "acctestaks%d"
+	node_resource_group = "ExampleResourceGroup"
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_kubernetes_cluster_test.go]: to protect against this running in parallel could we make this:

```suggestion
  node_resource_group = "acctestRGAKS-%d"
```

(and then thread through another rInt below, to ensure that this resource group is unique?)

**Comment by tombuildsstuff** [comment]
@infuerno nope, all good (I ended up re-ordering the file whilst passing through, nothing major) - I've just been waiting for the tests to run - once they've passed we're good to merge 👍

**Comment by tombuildsstuff** [comment]
Tests look good - thanks again for this @invidian :)

**Pull request merged**